File: UI/OpticSoftwareUI/src/app/core/entities/company/companyEntity.ts
import { BaseEntity } from '../baseEntity';
import { UserDetailEntity } from '../user/userDetailEntity';
import { CompanyParameterEntity } from './companyParameterEntity';

export class CompanyEntity extends BaseEntity {
  name: string;
  expireDate: Date;
  isActive: boolean;
  smsCount: number;
  userDetails: UserDetailEntity[];
  companyParameters: CompanyParameterEntity[];
}

File: OpticSoftware.Entity/Entities/Company/CompanyEntity.cs
using OpticSoftware.Entity.Entities.Company;
using OpticSoftware.Entity.Entities.User;
using System;
using System.Collections.Generic;
using System.Text;

namespace OpticSoftware.Entity.Entities.Company
{
    public class CompanyEntity : BaseEntity
    {
        public string Name { get; set; }
        public DateTime ExpireDate { get; set; }
        public bool IsActive { get; set; }
        public int SMSCount { get; set; }

        public virtual List<UserDetailEntity> UserDetails { get; set; }
        public virtual List<CompanyParameterEntity> CompanyParameters { get; set; }
    }
}
